src/lib.rs

```diff
@@ -94,59 +113,129 @@ fn write_mod_cmd<W: Write>(config: &ModConfig, ctx: &ConfigContext, mut writer:
+                            let res = Command::new(&cmd)
+                                .arg("-cybercmd")
+                                .current_dir(ctx.game_dir)
+                                .creation_flags(NO_WINDOW_FLAGS)
+                                .status()
+                                .ok();
+                            if !matches!(res, Some(st) if st.success()) {
+                                log.error(format!("Task \"{}\" failed when run.", task.command));
+                                if task.terminate_on_errors {
+                                    std::process::exit(0);
+                                }
+                            }
+                        } else {
+                            log.error(format!("Task \"{}\" in invalid location.", task.command));
+                            if task.terminate_on_errors {
+                                std::process::exit(0);
+                            }
+                        }
+                    } else {
+                        log.error(format!(
+                            "Task \"{}\" not found. ({:?})",
+                            task.command,
+                            dunce::simplified(cmd_path.as_path())
+                        ));
+                        if task.terminate_on_errors {
+                            std::process::exit(0);
+                        }
+                    }
                 }
             }
         }
     }
     Ok(())
 }
 
-fn get_game_path() -> Result<PathBuf> {
-    let exe = std::env::current_exe()?;
-    let path = exe.parent().unwrap().parent().unwrap().parent().unwrap();
-    Ok(path.to_path_buf())
-}
-
-fn get_configs() -> Result<Vec<ModConfig>> {
-    let path = get_game_path()?.join("r6").join("config").join("cybercmd");
+pub fn get_configs(log: &mut Logger) -> Result<Vec<ModConfig>> {
+    let path = paths::get_game_path()?
+        .join("r6")
+        .join("config")
+        .join("cybercmd");
     let mut configs = vec![];
 
-    for entry in std::fs::read_dir(path)? {
+    log.debug("Getting configs.");
+
+    for entry in fs::read_dir(path)? {
         let entry = entry?;
         if entry.path().extension() == Some(OsStr::new("toml")) {
-            let contents = std::fs::read_to_string(entry.path())?;
-            configs.push(toml::from_str(&contents)?);
+            log.debug(format!(
+                "Loading: {}",
+                dunce::simplified(&entry.path()).display()
+            ));
+            let contents = fs::read_to_string(entry.path())?;
+            match toml::from_str(&contents) {
+                Ok(config) => configs.push(config),
+                Err(error) => log.error(format!(
+                    "In {} ({}): {}",
+                    dunce::simplified(&entry.path()).display(),
+                    match error.span() {
+                        Some(val) => format!("{:?}", val),
+                        None => "".to_string(),
```

Comment:
I'd rather write String::new(), since it's statically guaranteed not to allocate